app/src/main/java/com/zulip/android/filters/NarrowFilterPM.java

```diff
@@ -73,8 +73,12 @@ public boolean matches(Message msg) {
     public String getTitle() {
         ArrayList<String> names = new ArrayList<>();
         for (Person person : people) {
-            if (!person.equals(ZulipApp.get().getYou()) || people.size() == 1) { //If PM to self then show title as your name
```

Comment:
hmm... I think the comment here indicates that if `people.size()` is one, it means the narrow is with the person itself. The only problem is when user narrows to itself, the people list has two person objects both pointing to user itself. This is because https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/activities/ZulipActivity.java#L1686 is missing a condition where if `person.equals(app.getYou())` then we might just want to pass an arraylist with one person object i.e app.getYou(). So, I think we could reduce all this code to just an if condition in narrow_with_pm().